Sources/PackageModel/UserToolchain.swift

```diff
@@ -328,7 +328,11 @@ public final class UserToolchain: Toolchain {
         swiftSDK: SwiftSDK,
         environment: EnvironmentVariables
     ) throws -> [String] {
-        let swiftCompilerFlags = swiftSDK.toolset.knownTools[.swiftCompiler]?.extraCLIOptions ?? []
+        var swiftCompilerFlags = swiftSDK.toolset.knownTools[.swiftCompiler]?.extraCLIOptions ?? []
+
+        if let linker = swiftSDK.toolset.knownTools[.linker]?.path {
+            swiftCompilerFlags += ["--ld-path=\(linker)"]
```

Comment:
Why not use `-use-ld=\(linker)` here?  I'd really rather not introduce new spellings for existing functionality.

Reply:
@kabiroberai I had a chat about this offline and we landed on `-ld-path`, I've submitted follow up PRs for that:

https://github.com/apple/swift/pull/68495
https://github.com/apple/swift-driver/pull/1441

There's a still a question what happens when both options are passed as `-use-ld=silly-linker -ld-path=/usr/bin/funny-linker`. We also need a test to verify that whatever behavior is when both of those options are passed doesn't regress.

Reply:
thanks for handling the upstream side of things! I believe `-use-ld` and `-ld-path` are (mostly) orthogonal but I see how that could regress; that test would be best suited to swift-driver right?

Reply:
Yes, tests for this are more appropriate for Swift Driver I believe. cc @artemcm